### DISCUSSION TITLE
Updating error looks like prior version issue:  Looking for a non-git-nerd solution

### DISCUSSION THREAD

**Discussion by soillives** [ohmyzsh/ohmyzsh#10344] [Help] [answered]
Here is what happened and my questions.  There is some background at bottom.  Thanks for helping to explain and fix this small problem.

I last updated OMZ successfully on 9 September.  I've tried to update several times since and always get  same error, here is the gist of it:

````
error: Pulling is not possible because you have unmerged files.
hint:...
fatal: Exiting because of an unresolved conflict.
````

Typing "git status" from $ZSH (~/.oh-my-zsh) shows the problem:

````
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	deleted:    plugins/autojump/README.md

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add/rm <file>..." as appropriate to mark resolution)
	deleted by them: plugins/autojump/autojump.plugin.zsh
````

I didn't change that README.md file nor did any of my processes.  So it was apparently embedded within the prior update, or perhaps the prior update did not finish successfully.

**MY QUESTIONS:**

- Should I assume that some OMZ updates are buggy and be prepared to ditch them?
- When faced with the unable-to-update situation above, is there an easy fix that covers all these kinds of buggy updates?
- Should I be thinking differently about this?

In the present case, I simply want the newer version of OMZ that was advertised.  I don't want to try to figure out what (outside my control) went wrong with the prior update.

I'll use a git-based solution but I hope I don't have to know how  git works in order to use OMZ.

**ASSUME**:
Assume that I'm ready to throw away any OMZ update that appears to be getting in the way, even reverting to a prior version if a newer one is not available.
Assume that I will do a routine check for changes under ZSH/custom/ to see if I can fix something.

**BACKGROUND**:
I like OMZ and hope to keep using it as troublefree as possible.  I use it on MacOS Big Sur (latest).

I like the idea of git, but I don't know much about it, and will not be learning it soon.  I just want to use OMZ without too many hiccups.

I'm comfortable at the Unix command line.  

I installed OMZ a few months ago.  I've made one tiny customization, it gives no trouble and sits where it should in the $ZSH/custom/ directory.

**Comment by mcornella** [comment, marked as answer]
Looks like some changes were made to the `autojump` plugin. As a rule of thumb, if you haven't made any changes to the OMZ folder that you need to preserve, you can just `git reset --hard` them and the update should proceed without issues:

```zsh
git -C "$ZSH" reset --hard HEAD
```

This will not make any changes to files in ignored paths, such as `$ZSH/custom`.

**Reply by soillives** [reply]
From $HOME I issued "git -C "$ZSH" reset --hard HEAD" then opened a new session/tab (in Terminal) and said "Y" to the Would you like to update? prompt.

The update now completes normally.

Thanks for confirming the likelihood that issue located by "git status" was the source of trouble.  And for giving a recovery command _and rationale behind it_ for situations like this.  I appreciate it, and likewise all the past and continuing of efforts of folks who've designed and created OMZ.  It's given my transition from bash to zsh a sweet taste.
